From d723677b15e5e606306ede9f69e2828939b48e9c Mon Sep 17 00:00:00 2001
From: shuai Wang <wangshuai12@xiaomi.corp-partner.google.com>
Date: Thu, 12 Mar 2026 21:06:30 +0800
Subject: [PATCH 08/63] ANDROID: GKI: update xiaomi symbol list

3 function symbol(s) added
  'u64 __blkg_prfill_u64(struct seq_file*, struct blkg_policy_data*, u64)'
  'void blkg_conf_exit(struct blkg_conf_ctx*)'
  'void blkg_conf_init(struct blkg_conf_ctx*, char*)'

Bug: 492104687
Bug: 495352351

Change-Id: I210f0351950e20f09288debdd3298f28291e4012
Signed-off-by: shuai Wang <wangshuai12@xiaomi.corp-partner.google.com>
(cherry picked from commit d9e7ad6b2d4f6f2a2d7c50e291394e1ae648af3c)
---
 gki/aarch64/abi.stg        | 48 ++++++++++++++++++++++++++++++++++++++
 gki/aarch64/symbols/xiaomi |  7 ++++++
 2 files changed, 55 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index 5fc2770e6621..f9aec4147f73 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -379505,6 +379505,12 @@ function {
   parameter_id: 0x0beab59b
   parameter_id: 0x18bd6530
 }
+function {
+  id: 0x1201f8b1
+  return_type_id: 0x48b5725f
+  parameter_id: 0x0a2e9ae5
+  parameter_id: 0x0483e6f8
+}
 function {
   id: 0x12051b11
   return_type_id: 0x48b5725f
@@ -379553,6 +379559,11 @@ function {
   parameter_id: 0x0c56e1ee
   parameter_id: 0x6720d32f
 }
+function {
+  id: 0x1213f72a
+  return_type_id: 0x48b5725f
+  parameter_id: 0x0a2e9ae5
+}
 function {
   id: 0x121493f8
   return_type_id: 0x48b5725f
@@ -395021,6 +395032,13 @@ function {
   parameter_id: 0x4585663f
   parameter_id: 0xf1a6dfed
 }
+function {
+  id: 0x56fc975d
+  return_type_id: 0x92233392
+  parameter_id: 0x0665e6b6
+  parameter_id: 0x26ad5dc1
+  parameter_id: 0x92233392
+}
 function {
   id: 0x56fd367d
   return_type_id: 0x18bd6530
@@ -435421,6 +435439,15 @@ elf_symbol {
   type_id: 0x56d6c623
   full_name: "__blkg_prfill_rwstat"
 }
+elf_symbol {
+  id: 0xbf1d0365
+  name: "__blkg_prfill_u64"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x12bcedf6
+  type_id: 0x56fc975d
+  full_name: "__blkg_prfill_u64"
+}
 elf_symbol {
   id: 0x2d1c1f2d
   name: "__blockdev_direct_IO"
@@ -458012,6 +458039,24 @@ elf_symbol {
   type_id: 0x9dfe50e2
   full_name: "blkdev_issue_zeroout"
 }
+elf_symbol {
+  id: 0x3a816bcd
+  name: "blkg_conf_exit"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x9b2aecc6
+  type_id: 0x1213f72a
+  full_name: "blkg_conf_exit"
+}
+elf_symbol {
+  id: 0xb0f09bc0
+  name: "blkg_conf_init"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xc09f033d
+  type_id: 0x1201f8b1
+  full_name: "blkg_conf_init"
+}
 elf_symbol {
   id: 0x35eb35db
   name: "blkg_conf_prep"
@@ -525960,6 +526005,7 @@ interface {
   symbol_id: 0x5a6afeee
   symbol_id: 0xc8eb813d
   symbol_id: 0x7bf707d4
+  symbol_id: 0xbf1d0365
   symbol_id: 0x2d1c1f2d
   symbol_id: 0x9bed4545
   symbol_id: 0xefdcf4c1
@@ -528470,6 +528516,8 @@ interface {
   symbol_id: 0x4d722bb5
   symbol_id: 0x2a322266
   symbol_id: 0x8c705d8e
+  symbol_id: 0x3a816bcd
+  symbol_id: 0xb0f09bc0
   symbol_id: 0x35eb35db
   symbol_id: 0x120ecf6c
   symbol_id: 0xabd39b51
diff --git a/gki/aarch64/symbols/xiaomi b/gki/aarch64/symbols/xiaomi
index 5c6b18681173..d32be247183a 100644
--- a/gki/aarch64/symbols/xiaomi
+++ b/gki/aarch64/symbols/xiaomi
@@ -215,6 +215,13 @@
   blk_mq_sched_mark_restart_hctx
   blk_mq_sched_try_insert_merge
   elv_rqhash_add
+  blkcg_print_blkgs
+  blkg_conf_init
+  blkg_conf_prep
+  blkg_conf_exit
+  __blkg_prfill_u64
+  blkcg_root
+  blkcg_activate_policy
 
 # required by mi_damon.ko
   folio_deactivate
-- 
2.50.1

